scripts/benchmarks_local.py

```diff
@@ -0,0 +1,451 @@
+    parser.add_argument('--os', choices=['windows', 'linux'], default=platform.system().lower(), help='Specifies the operating system of the system')
+    parser.add_argument('--filter', type=str, default='*', help='Specifies the benchmark filter to pass to BenchmarkDotNet')
+    parser.add_argument('-f', '--framework', choices=ChannelMap.get_supported_frameworks(), default='net8.0', help='The target framework to run the benchmarks against.') # Can and should this accept multiple frameworks?
+    parser.add_argument('--csproj', type=str, default=os.path.join("..", "src", "benchmarks", "micro", "MicroBenchmarks.csproj"), help='The path to the csproj file to run benchmarks against.')    
+
+def __main(args: list):
+    # Define the ArgumentParser
+    parser = ArgumentParser(description='Run local benchmarks for the Performance repo.')
+    add_arguments(parser)
+    parsed_args = parser.parse_args(args)
+    
+    setup_loggers(verbose=parsed_args.verbose)
+
+    # Ensure we are running as admin
+    if not is_running_as_admin(parsed_args):
+        if parsed_args.allow_non_admin_execution:
+            getLogger().warning("This script is not running as an administrator. This may cause errors.")
+        else:
+            raise Exception("This script must be run as an administrator or --allow-non-admin-execution must be passed.")
+
+    # If list cached builds is specified, list the cached builds and exit
+    if parsed_args.list_cached_builds:
+        for folder in os.listdir(parsed_args.artifact_storage_path):
+            if any([run_type.name in folder for run_type in RunType]):
+                getLogger().info(folder)
+        return
+
+    # Check to make sure we have something specified to test
+    if parsed_args.commits or parsed_args.local_test_repo:
+        if parsed_args.commits:
+            getLogger().info(f"Commits to test are: {parsed_args.commits}")
+        if parsed_args.local_test_repo:
+            getLogger().info(f"Local repo to test is: {parsed_args.local_test_repo}")
+    else:
+        raise Exception("A commit id and/or local repo must be specified.")
+
+    getLogger().debug(f"Input arguments: {parsed_args}")
+
+    repo_dirs = []
+    repo_url = parsed_args.repo_url
+    if parsed_args.commits:
+        check_references_exist_and_add_branch_commits(repo_url, parsed_args.commits, parsed_args.repo_storage_path, repo_dirs[0] if parsed_args.separate_repos else "runtime")
+        for commit in parsed_args.commits:
+            repo_dirs.append(f"runtime-{commit.replace('/', '-')}")
+
+    try:
+        getLogger().info("Killing any running dotnet, vstest, or msbuild processes... (ignore system cannot find path specified)")
+        kill_dotnet_processes(parsed_args)
+
+        # Generate the artifacts for each of the remote versions
+        if parsed_args.commits:
+            getLogger().info(f"Checking if references {parsed_args.commits} exist in {repo_url}.")
```

Comment:
Should this go on line 413 when we are actually checking if these references exist?